src/sketch.lisp

```diff
@@ -260,7 +288,28 @@
 ;;; Control flow
 
 (defun stop-loop ()
-  (setf (sdl2.kit:idle-render *sketch*) nil))
+  (setf (sdl2.kit:idle-render (sketch-%window *sketch*)) nil))
 
 (defun start-loop ()
-  (setf (sdl2.kit:idle-render *sketch*) t))
+  (setf (sdl2.kit:idle-render (sketch-%window *sketch*)) t))
+
+;; For backward compatibility.
+(defmethod kit.sdl2:idle-render ((instance sketch))
+  (kit.sdl2:idle-render (sketch-%window instance)))
+
+(defmethod (setf kit.sdl2:idle-render) (value (instance sketch))
+  (setf (kit.sdl2:idle-render (sketch-%window instance)) value))
+
+;;; Resource-handling
+
+(defun delay-init-p ()
+  "This checks whether the OpenGL context has been created yet. If not,
+we need to wait before initializing certain resources."
+  (and *sketch*
+       (null (sketch-%window *sketch*))))
+
+(defun add-delayed-init-fun! (f)
+  "F should be a function with no arguments."
+  (setf (sketch-%delayed-init-funs *sketch*)
+        (append (sketch-%delayed-init-funs *sketch*)
```

Comment:
It doesn't matter in which order to call `%delayed-init-funcs`, and using `(push f (sketch-%delayed-init-funs *sketch*))` would be more efficient.

If you really want to keep the order, you could do `(reverse (sketch-%delayed-init-funs))` in `initialize-instance`.

Reply:
The order could potentially matter if we add other types of delayed initializations. And calling `reverse` in another place makes the code more difficult to understand. So I'd prefer to leave as-is. If there's ever a case where someone is creating 1000s of OpenGL resources and efficiency becomes an issue, then we can always change it.

Reply:
I still think that current implementation of `add-delayed-init-fun!` conses too much.

To keep the code understandable you could rename the slot to for example `%reversed-delayed-init-funs`.
If you don't like the idea of using `push` + `reverse` / `nreverse`; there are other solutions to this inefficiency, for example you could use a queue (although you'd need to implement one in utils, I guess) or a vector (with `vector-push-extend`).

Reply:
Okay, I'm happy to switch to a vector if you feel strongly about it! :+1: